### ISSUE TITLE
Last Gridcell/Sub routing question

### ISSUE THREAD

**Issue by mee067** [MESH-Model/MESH-Dev#36]
This is just a question. MESH used not to run the LSS for the very last gridcell that has NEXT=0. Newer code, not sure since which release, started to run the LSS for that last gridcell. To make things equivalent to old code, one would set all GRU fractions for that cell to zero for the LSS not to run. Maybe this is related to vector setups as the workflow does not add that additional sub-basin which will have a Rank with NEXT=0 as GK used to do.

I know that last gridcell was added to enable adding its runoff and routing it to the outlet with all the incoming flow.

Now the LSS will generate runoff for that last grid or sub which is not an extra. Does the routing still work for it while its NEXT = 0?

**Comment by dprincz** [comment]
The routing will only run over `1:NAA`, but it will put the outflow from `NAA` into `NA`. I don't believe runoff over `NA` is routed (meaning `qo` only includes `qi` from upstream cells). The way to do that is to set `NAA=NA`. It'll print a warning if that's the case, but I don't believe it will halt the run (I think setups ported from GEM-Hydro usually have the condition NAA=NA).

This is a good consideration, as runoff over the downstream-most subbasin in vector setups might not see full routing (e.g., incorporating runoff over that subbasin itself). It could be something for someone to investigate.

**Comment by mee067** [comment]
From what I gather, this setting of NAA=NA would be inside the code, right?

For large basins that have one outlet at the end, runoff from the last grid/sub may not add much but for smaller basins, and for a setup with multiple outlets, excluding runoff of the last grid/sub or not routing it may be significant.

**Comment by dprincz** [comment]
I'm not sure `NAA=NA` should be an assumption set by the code. It could be set by the fields in the drainage database (e.g., by having no cell with `NEXT==0`.

**Comment by sujata91** [comment]
Just checking in, has there been any update on resolving this issue, either in the meshflow or the MESH code itself? Please let me know, as I now have multiple outlet locations in my vector-based setup.

Thanks!

**Comment by mee067** [comment]
For grid-based setups. Green Kenue used to add a grid below the outlet (I think this comes all the way from WATFLOOD) which had no value except for achieving the above-mentioned purpose. Earlier versions of MESH used not to run LSS for that grid even if it had GRU fractions. This was changed a while ago, not sure why and it initially caused some issues for old setups where the cell had GRU fractions but no other properties and I had to go set the GRU fractions to zero to replicate the original behaviour. Things have to be compatible between grid-based and vector-based setups, whether an additional grid-cell or subbasin are added in the setup workflow or by MESH. If this cell/sub is fictious, we needn't worry about runoff and flow from it.

**Comment by mee067** [comment]
I am saying so because if MESH adds this internally, it may need to check whether the setup already has that extra cell. Most likely it will, if GK was used for development. I do not know if Zel's R scripts that can replace GK for grid-based setups does the same thing or not.

**Comment by dprincz** [comment]
Agreed. I was thinking it would have to be tied to some check. For those setups converted from fst/fstd standard files for example, the check is easy because the next of the outlet equals its rank, which already triggers a warning.

The alternative I was considering is it should be forced by an option on `SHDFILEFLAG` than by letting the code decide based on what it interprets from the indexing after reading the files.
